clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -194,12 +194,25 @@ public MemoryRecords build() {
      * @return The max timestamp and its offset
      */
     public RecordsInfo info() {
-        if (timestampType == TimestampType.LOG_APPEND_TIME)
-            return new RecordsInfo(logAppendTime,  lastOffset);
-        else if (maxTimestamp == RecordBatch.NO_TIMESTAMP)
+        if (timestampType == TimestampType.LOG_APPEND_TIME) {
```

Comment:
The implementation looks right. Do we need to change the comment above the method accordingly?